nexus-repository-r/src/test/java/org/sonatype/nexus/repository/r/internal/upload/RUploadHandlerTest.java

```diff
@@ -155,10 +157,39 @@ public void testHandle_unauthorized() throws IOException {
     }
   }
 
-  private ComponentUpload createComponentUpload(final String packagePath) {
+  @Test
+  public void testHandleValidationExceptionWrongPath() throws IOException {
+    ComponentUpload component = createComponentUpload(WRONG_PACKAGE_PATH, PACKAGE_NAME);
+    try {
+      underTest.handle(repository, component);
+      fail("Expected validation exception");
+    }
+    catch (ValidationErrorsException e) {
+      assertThat(e.getValidationErrors().size(), is(1));
+      assertThat(e.getValidationErrors().get(0).getMessage(),
+          is("Non-R archive extension or wrong upload path."));
```

Comment:
do we need to check exception error in unit-tests?

Reply:
yes, cause we have some different validation with different messages and I need to make sure I caught the right one